lib/pecorino.rb

```diff
@@ -1,11 +1,17 @@
 # frozen_string_literal: true
 
+require "active_support/concern"
+require "active_record/sanitization"
+
 require_relative "pecorino/version"
 require_relative "pecorino/leaky_bucket"
 require_relative "pecorino/throttle"
 require_relative "pecorino/railtie" if defined?(Rails::Railtie)
 
 module Pecorino
+  autoload :Postgres, "pecorino/postgres"
```

Comment:
**suggestion:** Thanks for using `autoload` here instead of `require`. It's a great improvement.

But I'd like to suggest taking it even further and use `zeitwerk`. We have so many rails dependencies for this gem, that it's safe to assume, that `zeitwek` will be available in most environments where this gem is installed.

Zeitwerk offers some significant improvements, besides lazy-loading, there is a good summary of those [here](https://discourse.dry-rb.org/t/what-are-the-differences-between-the-default-loader-and-the-zeitwerk-loader/1462/2). It seems like a worthy change.

Reply:
We can probably move this into separate issue.